intercepting-filter/src/main/App.java

```diff
@@ -0,0 +1,245 @@
+ import java.util.*;
+import javax.swing.*;
+import javax.swing.table.*;
+import java.awt.*;
+import java.awt.event.*;
+public class App{
+	public static void main(String[] args){
+		FilterManager filterManager = new FilterManager(new Target());
+		filterManager.setFilter(new NameFilter());
+		filterManager.setFilter(new ContactFilter());
+		filterManager.setFilter(new AddressFilter());
+		filterManager.setFilter(new DepositFilter());
+		filterManager.setFilter(new OrderFilter());
+
+		Client client = new Client();
+		client.setFilterManager(filterManager);
+	}
+}
+
+interface Filter{
+	public String execute(String[] request);
+}
+
+class NameFilter implements Filter{
+	public String execute(String[] request){
+		if(request[0].equals("") || request[0].matches(".*[^\\w|\\s]+.*")){
+			return null;
+		}else return request[0];
+	}
+}
+
+class ContactFilter implements Filter{
+	public String execute(String[] request){
+		if(request[1].equals("") || request[1].matches(".*[^\\d]+.*") || request[1].length() != 11){
+			return null;
+		}else return request[1];
+	}
+}
+
+class AddressFilter implements Filter{
+	public String execute(String[] request){
+		if(request[2].equals("")){
+			return null;
+		}else return request[2];
+	}
+}
+
+class DepositFilter implements Filter{
+	public String execute(String[] request){
+		if(request[3].equals("")){
+			return null;
+		}else return request[3];
+	}
+}
+
+class OrderFilter implements Filter{
+	public String execute(String[] request){
```

Comment:
There is couple of Filter implementation that has almost the same code.
Can we eliminate them whit one implementaion and inject only variable that varies through constructor?